web-ui/src/components/admin/users/Users.jsx

```diff
@@ -69,6 +69,31 @@ const Users = () => {
       ? selectNormalizedMembersAdmin(state, searchText)
       : selectNormalizedMembers(state, searchText);
 
+  useEffect(() => {
+    const url = new URL(location.href);
+
+    const addUser = url.searchParams.get('addUser');
+    setOpen(addUser === 'true');
+
+    const includeTerminated = url.searchParams.get('includeTerminated');
+    setIncludeTerminated(includeTerminated === 'true');
+
+    const search = url.searchParams.get('search') || '';
+    setSearchText(search);
+  }, []);
+
+  useEffect(() => {
```

Comment:
Every time a piece of the page state changes, this updates the query parameters in the URL.